### ISSUE TITLE
Broken manifest when adding Stitchport to node with multiple existing interfaces

### ISSUE THREAD

**Issue by paul-ruth** [RENCI-NRIG/orca5#223]
This error is seen with the CICI-SAFE sdx code.  The slice has nodes at multiple sites connected with interdomain broadcast networks.  Each site also has additional node(s) connected with an intradomain broadcast networks.  None of the nodes have IPs.  

If we add a stitchport to one of the nodes that has multiple links the slice works (I think) but the manifest is broken.

An example broken manifest is attached.

[cici-safe-broken-manifest.txt](https://github.com/RENCI-NRIG/orca5/files/2401499/cici-safe-broken-manifest.txt)

**Comment by paul-ruth** [comment]
Here is a set of requests and manifests that can be use to recreate the issue.  Note they all end in txt (b/c github)


[example-broken-request-manifest2.txt](https://github.com/RENCI-NRIG/orca5/files/2401547/example-broken-request-manifest2.txt)
[example-broken-request-manifest1.txt](https://github.com/RENCI-NRIG/orca5/files/2401548/example-broken-request-manifest1.txt)
[example-broken-request.txt](https://github.com/RENCI-NRIG/orca5/files/2401549/example-broken-request.txt)

**Comment by kthare10** [comment]
File attached is for a smaller slice.
Scenario:
A) Slice created with 3 Nodes Node1, Node 2 and Node 3 with both Node 1 and Node 3 connected to Node 2
B) Slice modified to add Stitch Port SP1 to Node 2, manifest fails to load after the modify

[ahab-modify-SP1.txt](https://github.com/RENCI-NRIG/orca5/files/2456682/ahab-modify-SP1.txt)

**Comment by kthare10** [comment]
Test: Perform Stateful restart controller after modify
Observation: Manifest is loaded successfully after stateful restart of the controller.
Subsequent modify still fails with the same error

Attaching manifest after the stateful restart
[ahab-modify-SP1-afterrestart.txt](https://github.com/RENCI-NRIG/orca5/files/2456886/ahab-modify-SP1-afterrestart.txt)

**Comment by ibaldin** [comment]
What is the actual error that is received (and where?) @kthare10

**Comment by kthare10** [comment]
Scenario:
Create a slice with 3 Nodes N1, N2 and N3 with links as N1-N2-N3
Modify slice by adding stitchport to N2
Modify is successful but resulting manifest is broken

Polling Manifest view from flukes results in below exception:
![screen shot 2018-10-10 at 12 44 48 pm](https://user-images.githubusercontent.com/36123970/46775400-16f42700-ccd5-11e8-963a-715a6f7d278d.png)

**Comment by kthare10** [comment]
Exception is observed irrespective of nodes being on single domain or inter domain

**Comment by ibaldin** [comment]
The path from stitchport SP1 ends without reaching the stitchport node, after the LinkConnection object with interface `http://geni-orca.renci.org/owl/orca.rdf#dbf6a192-8b4a-4a1b-b4c3-8089d2536982/Stitching/Domain/intf`. 

LinkConnection:
```
  <rdf:Description rdf:about="http://geni-orca.renci.org/owl/ion.rdf#AL2S/Chameleon/Cisco/6509/GigabitEthernet/1/1/3292-dbf6a192-8b4a-4a1b-b4c3-8089d2536982/Stitching/Domain/intf/92ec909a-dab7-44bf-a758-45e2e0971a1d">
    <rdf:type rdf:resource="http://geni-orca.renci.org/owl/topology.owl#LinkConnection"/>
    <j.14:hasInterface rdf:resource="http://geni-orca.renci.org/owl/ion.rdf#AL2S/Chameleon/Cisco/6509/GigabitEthernet/1/1/3292"/>
    <j.14:hasInterface rdf:resource="http://geni-orca.renci.org/owl/orca.rdf#dbf6a192-8b4a-4a1b-b4c3-8089d2536982/Stitching/Domain/intf"/>
  </rdf:Description>
```

The LinkConnection leads back to ION:
```
  <rdf:Description rdf:about="http://geni-orca.renci.org/owl/ion.rdf#ion/Domain/vlan/5590af42-f88d-4290-b89e-580b6920f549/vlan">
    <rdf:type rdf:resource="http://geni-orca.renci.org/owl/topology.owl#CrossConnect"/>
    <j.2:inDomain rdf:resource="http://geni-orca.renci.org/owl/ion.rdf#ion/Domain/vlan"/>
    <j.2:message>Reservation 184714dc-265e-40bb-a3f5-57441c64a09d (Slice kthare10.slice1) is in state [Nascent,Ticketing]
</j.2:message>
    <j.2:hasReservationState rdf:resource="http://geni-orca.renci.org/owl/request.owl#Nascent"/>
    <j.14:hasInterface rdf:resource="http://geni-orca.renci.org/owl/ion.rdf#ION/StarLight/Cisco/6509/TenGigabitEthernet/1/1/ethernet"/>
    <j.14:hasInterface rdf:resource="http://geni-orca.renci.org/owl/ion.rdf#AL2S/Chameleon/Cisco/6509/GigabitEthernet/1/1/3292"/>
    <j.7:hasResourceType rdf:resource="http://geni-orca.renci.org/owl/domain.owl#VLAN"/>
    <j.14:hasURL>http://geni-orca.renci.org/owl/ion.rdf#ion/Domain/vlan/5590af42-f88d-4290-b89e-580b6920f549/vlan</j.14:hasURL>
    <j.10:inRequestNetworkConnection rdf:resource="http://geni-orca.renci.org/owl/84f9cbaf-d519-4ac0-9a98-967c5333a5ef#SP1-net"/>
    <j.14:inConnection rdf:datatype="http://www.w3.org/2001/XMLSchema#boolean">true</j.14:inConnection>
  </rdf:Description>
```

Stitchport SP1 has no interfaces: 
```
  <rdf:Description rdf:about="http://geni-orca.renci.org/owl/84f9cbaf-d519-4ac0-9a98-967c5333a5ef#SP1">
    <rdf:type rdf:resource="http://geni-orca.renci.org/owl/topology.owl#Device"/>
    <j.2:inDomain rdf:resource="http://geni-orca.renci.org/owl/orca.rdf#dbf6a192-8b4a-4a1b-b4c3-8089d2536982/Stitching/Domain"/>
    <j.14:hasURL>http://geni-orca.renci.org/owl/orca.rdf#dbf6a192-8b4a-4a1b-b4c3-8089d2536982/Stitching/Domain</j.14:hasURL>
    <j.10:inRequestNetworkConnection rdf:resource="http://geni-orca.renci.org/owl/84f9cbaf-d519-4ac0-9a98-967c5333a5ef#SP1-net"/>
    <j.14:inConnection rdf:datatype="http://www.w3.org/2001/XMLSchema#boolean">true</j.14:inConnection>
  </rdf:Description>
```

The path up to the link connection from the VM appears valid. The problem looks like the crossconnect node corresponding to the domain of the Chameleon stitchport and the associated interfaces are missing.

**Comment by ibaldin** [comment]
Simplified manifest attached for the record. 

[sp-1-simplified.txt](https://github.com/RENCI-NRIG/orca5/files/2469648/sp-1-simplified.txt)

**Comment by kthare10** [comment]
Embedding code is generating the same path for success(create) as well as failure(modify) scenario.
```
2018-10-15 13:17:52,945 [qtp115126871-778 - /orca/xmlrpc] DEBUG ndl.logger - Shortest path solution: [
[http://geni-orca.renci.org/owl/orca.rdf#e863068c-ccdb-4450-ad26-e9c0c33b07ce/Stitching/Domain, http://geni-orca.renci.org/owl/orca.rdf#e863068c-ccdb-4450-ad26-e9c0c33b07ce/Stitching/Domain/intf, -1],
[http://geni-orca.renci.org/owl/ionradics.rdf#ionradics/Domain/vlan, http://geni-orca.renci.org/owl/ionradics.rdf#AL2S/UIUC/Cisco/6509/GigabitEthernet/1/1/1141, 0],
[http://geni-orca.renci.org/owl/ionradics.rdf#ionradics/Domain/vlan, http://geni-orca.renci.org/owl/ionradics.rdf#AL2S/Renci/Cisco/6509/TenGigabitEthernet/1/1, 10000000000],
[http://geni-orca.renci.org/owl/disaNet.rdf#disaNet/Domain/vlan, http://geni-orca.renci.org/owl/disaNet.rdf#DisaNet/IBM/G8264/TenGigabitEthernet/1/0/ethernet, -1],
[http://geni-orca.renci.org/owl/disaNet.rdf#disaNet/Domain/vlan, http://geni-orca.renci.org/owl/disaNet.rdf#DisaNet/IBM/G8264/TenGigabitEthernet/1/1/ethernet, -1],
[http://geni-orca.renci.org/owl/disavmsite.rdf#disavmsite/Domain/vm, http://geni-orca.renci.org/owl/disavmsite.rdf#Disa/ExoGeni/TenGigabitEthernet/1/ethernet, -1]
]
```

**Comment by kthare10** [comment]
Scenario:
1) Create a slice containing 1 VM
2) Modify slice by attaching a stitch port
3) Modify is successful but the polling the manifest results in an exception 
4) Subsequent modifies to the slice fail

NOTE: Similar scenario where VM and stitch port are created together at the create slice works.

Investigation and comparison of these scenarios pointed out that the Path computed for both is same by the embedding logic:

```
2018-10-18 13:55:55,376 [qtp1067580739-47 - /orca/xmlrpc] DEBUG ndl.logger - OntProcessor::findShortestPath():Shortest path solution: 
[
[http://geni-orca.renci.org/owl/orca.rdf#a5f81b40-8ce2-4a2b-8b13-f5b04990b71a/Stitching/Domain, http://geni-orca.renci.org/owl/orca.rdf#a5f81b40-8ce2-4a2b-8b13-f5b04990b71a/Stitching/Domain/intf, -1], 
[http://geni-orca.renci.org/owl/ionradics.rdf#ionradics/Domain/vlan, http://geni-orca.renci.org/owl/ionradics.rdf#AL2S/UIUC/Cisco/6509/GigabitEthernet/1/1/1140, 0], 
[http://geni-orca.renci.org/owl/ionradics.rdf#ionradics/Domain/vlan, http://geni-orca.renci.org/owl/ionradics.rdf#AL2S/Renci/Cisco/6509/TenGigabitEthernet/1/1, 10000000000], 
[http://geni-orca.renci.org/owl/disaNet.rdf#disaNet/Domain/vlan, http://geni-orca.renci.org/owl/disaNet.rdf#DisaNet/IBM/G8264/TenGigabitEthernet/1/0/ethernet, -1], 
[http://geni-orca.renci.org/owl/disaNet.rdf#disaNet/Domain/vlan, http://geni-orca.renci.org/owl/disaNet.rdf#DisaNet/IBM/G8264/TenGigabitEthernet/1/1/ethernet, -1], 
[http://geni-orca.renci.org/owl/disavmsite.rdf#disavmsite/Domain/vm, http://geni-orca.renci.org/owl/disavmsite.rdf#Disa/ExoGeni/TenGigabitEthernet/1/ethernet, -1]
]
```
After this, both modify and create flow invokes CloudHandler:createEdgeDeviceCG(), which is responsible for creating the new NetworkConnection/Link. But following differences are observed. 

Input parameter to CloudHandler:createEdgeDeviceCG() is the ComputeElement and it does not have a resource map in case of modify. **Shouldn't the ComputeElement look same in case of create/modify???**

**ComputeElement at beginning of CloudHandler:createEdgeDeviceCG() for Create**
```
2018-10-18 13:55:56,489 [qtp1067580739-47 - /orca/xmlrpc] DEBUG ndl.logger - CloudHandler:createEdgeDeviceCG(): Type: ComputeElement
URL: http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Node0 Name: http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Node0 At Layer: null #sn:1 ResourceType: vm:1:10:http://geni-orca.renci.org/owl/disavmsite.rdf#disavmsite/Domain/vm ClientInterface: [
Type: Interface
URL: http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Link135-Node0 Name: http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Link135-Node0 At Layer: null #sn:null ResourceType: null ClientInterface: [
]
Interface Bandwidth:
Label:unspecified:null:IPAddress:null:http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Link135-Node0;cidr=null ]
Interface Bandwidth: [ memoryCapacity <> [memoryCapacity,bw=3000, tag=0, label listnull] ] [ numCPUCores <> [numCPUCores,bw=1, tag=0, label listnull] ] [ storageCapacity <> [storageCapacity,bw=25, tag=0, label listnull] ]
CE Type: xo.medium/false
Image: http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Centos+6.7+v1.1.0/http://geni-images.renci.org/images/standard/centos/centos6.7-v1.1.0/centos6.7-v1.1.0.xml/0c22c525b8a4f0f480f17587557b57a7a111d198
Group: null
nodeGroupName: null
iSCSI: null/null/null/null/false
```
**ComputeElement at beginning of CloudHandler:createEdgeDeviceCG() for Modify**
```
2018-10-18 14:09:18,313 [qtp1067580739-225 - /orca/xmlrpc] DEBUG ndl.logger - CloudHandler:createEdgeDeviceCG(): Type: ComputeElement
URL: http://geni-orca.renci.org/owl/747f797b-3a2c-4efe-ae06-9317c58bdbc6#Node1 Name: http://geni-orca.renci.org/owl/747f797b-3a2c-4efe-ae06-9317c58bdbc6#Node1 At Layer: null #sn:1 ResourceType: vm:1:10:http://geni-orca.renci.org/owl/disavmsite.rdf#disavmsite/Domain/vm ClientInterface: [
Type: Interface
URL: http://geni-orca.renci.org/owl/42ad8a60-aa45-4d91-b27e-8b9a4fa12e1f#Link140-Node1 Name: http://geni-orca.renci.org/owl/42ad8a60-aa45-4d91-b27e-8b9a4fa12e1f#Link140-Node1 At Layer: null #sn:null ResourceType: null ClientInterface: [
]
Interface Bandwidth:
Label:unspecified:null:IPAddress:null:http://geni-orca.renci.org/owl/42ad8a60-aa45-4d91-b27e-8b9a4fa12e1f#Link140-Node1;cidr=null ]
Interface Bandwidth:
CE Type: null/false
Image: null/null/null
Group: null
nodeGroupName: null
iSCSI: null/null/null/null/false
```
Also, in both cases, CloudHandler:createEdgeDeviceCG() invokes CloudHandler:createNewNode() to create the NetworkConnection and attaches it to ComputeElement. However, ComputeElement for modify at the end does not have the interface associated with it. CloudHandler:createNewNode() invokes setEdgeDependency() and adding debug traces in this function points out that the dependencies are differently generated for both modify and create.

**ComputeElement at completion of CloudHandler:createEdgeDeviceCG() for Create**
```
2018-10-18 13:55:56,497 [qtp1067580739-47 - /orca/xmlrpc] DEBUG ndl.logger - CloudHandler:createEdgeDeviceCG(): device:http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Node0;numInterface=0;isModify=false;modify version=0
2018-10-18 13:55:56,500 [qtp1067580739-47 - /orca/xmlrpc] DEBUG ndl.logger - CloudHandler:createEdgeDeviceCG(): Type: ComputeElement
URL: http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Node0 Name: http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Node0 At Layer: null #sn:1 ResourceType: vm:1:10:http://geni-orca.renci.org/owl/disavmsite.rdf#disavmsite/Domain/vm ClientInterface: [
Type: Interface
URL: http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Link135-Node0 Name: http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Link135-Node0 At Layer: null #sn:null ResourceType: null ClientInterface: [
]
Interface Bandwidth:
Label:unspecified:null:IPAddress:null:http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Link135-Node0;cidr=null Type: Interface
URL: http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Link135-Node0/0/intf Name: http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Link135-Node0/0/intf At Layer: null #sn:null ResourceType: null ClientInterface: [
]
Interface Bandwidth:
Label:unspecified:null:IPAddress:null:http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Link135-Node0/0;cidr=null ]
Interface Bandwidth: [ memoryCapacity <> [memoryCapacity,bw=3000, tag=0, label listnull] ] [ numCPUCores <> [numCPUCores,bw=1, tag=0, label listnull] ] [ storageCapacity <> [storageCapacity,bw=25, tag=0, label listnull] ]
CE Type: xo.medium/false
Image: http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Centos+6.7+v1.1.0/http://geni-images.renci.org/images/standard/centos/centos6.7-v1.1.0/centos6.7-v1.1.0.xml/0c22c525b8a4f0f480f17587557b57a7a111d198
Group: null
nodeGroupName: null
iSCSI: null/null/null/null/false
 OUT
```
**ComputeElement at completion of CloudHandler:createEdgeDeviceCG() for Modify**
```
2018-10-18 14:09:18,318 [qtp1067580739-225 - /orca/xmlrpc] DEBUG ndl.logger - CloudHandler:createEdgeDeviceCG(): device:http://geni-orca.renci.org/owl/747f797b-3a2c-4efe-ae06-9317c58bdbc6#Node1;numInterface=0;isModify=true;modify version=1
2018-10-18 14:09:18,318 [qtp1067580739-225 - /orca/xmlrpc] DEBUG ndl.logger - CloudHandler:createEdgeDeviceCG(): Type: ComputeElement
URL: http://geni-orca.renci.org/owl/747f797b-3a2c-4efe-ae06-9317c58bdbc6#Node1 Name: http://geni-orca.renci.org/owl/747f797b-3a2c-4efe-ae06-9317c58bdbc6#Node1 At Layer: null #sn:1 ResourceType: vm:1:10:http://geni-orca.renci.org/owl/disavmsite.rdf#disavmsite/Domain/vm ClientInterface: [
Type: Interface
URL: http://geni-orca.renci.org/owl/42ad8a60-aa45-4d91-b27e-8b9a4fa12e1f#Link140-Node1 Name: http://geni-orca.renci.org/owl/42ad8a60-aa45-4d91-b27e-8b9a4fa12e1f#Link140-Node1 At Layer: null #sn:null ResourceType: null ClientInterface: [
]
Interface Bandwidth:
Label:unspecified:null:IPAddress:null:http://geni-orca.renci.org/owl/42ad8a60-aa45-4d91-b27e-8b9a4fa12e1f#Link140-Node1;cidr=null ]
Interface Bandwidth:
CE Type: null/false
Image: null/null/null
Group: null
nodeGroupName: null
iSCSI: null/null/null/null/false
```

Subsequent processing for both create and modify scenarios invokes InterDomainHandler::createManifest(). Code flow for both create/modify is exactly identical but manifest at the completion of this function is missing hasInterface() from Link and StitchPort.

**Link and stitchport for Create after completion of InterDomainHandler::createManifest()**
```
  <rdf:Description rdf:about="http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Link135">
    <rdf:type rdf:resource="http://geni-orca.renci.org/owl/topology.owl#NetworkConnection"/>
    <j.9:hasInterface rdf:resource="http://geni-orca.renci.org/owl/ionradics.rdf#AL2S/UIUC/Cisco/6509/GigabitEthernet/1/1/1140"/>
    <j.9:hasInterface rdf:resource="http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Link135-Node0"/>
    <j.9:hasGUID>95d09426-8a4d-493c-a640-503490a4a7d4</j.9:hasGUID>
    <j.0:atLayer rdf:resource="http://geni-orca.renci.org/owl/ethernet.owl#EthernetNetworkElement"/>
    <j.0:bandwidth rdf:datatype="http://www.w3.org/2001/XMLSchema#integer">10000000</j.0:bandwidth>
    <j.8:item rdf:resource="http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#StitchPort0"/>
    <j.8:item rdf:resource="http://geni-orca.renci.org/owl/ionradics.rdf#AL2S/UIUC/Cisco/6509/GigabitEthernet/1/1/1140-a5f81b40-8ce2-4a2b-8b13-f5b04990b71a/Stitching/Domain/intf/a48daa48-b2c8-423a-af19-6f158807663f"/>
    <j.8:item rdf:resource="http://geni-orca.renci.org/owl/ionradics.rdf#ionradics/Domain/vlan/2cb2f14b-c31d-4011-8a92-3f5c9315cf1b/vlan"/>
  </rdf:Description>

  <rdf:Description rdf:about="http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#StitchPort0">
    <rdf:type rdf:resource="http://geni-orca.renci.org/owl/topology.owl#Device"/>
    <j.2:inDomain rdf:resource="http://geni-orca.renci.org/owl/orca.rdf#a5f81b40-8ce2-4a2b-8b13-f5b04990b71a/Stitching/Domain"/>
    <j.11:hasInterface rdf:resource="http://geni-orca.renci.org/owl/ionradics.rdf#AL2S/UIUC/Cisco/6509/GigabitEthernet/1/1/1140"/>
    <j.11:hasGUID>dc576d9f-3627-44c6-a13f-fb5dfe757791</j.11:hasGUID>
    <j.11:hasURL>http://geni-orca.renci.org/owl/orca.rdf#Stitching/Domain</j.11:hasURL>
    <j.7:inRequestNetworkConnection rdf:resource="http://geni-orca.renci.org/owl/62f7485a-8d0e-41cb-a3fb-e51979fd6525#Link135"/>
    <j.11:inConnection rdf:datatype="http://www.w3.org/2001/XMLSchema#boolean">true</j.11:inConnection>
  </rdf:Description>
```

**Link and stitchport for Modify after completion of InterDomainHandler::createManifest()**
```
  <rdf:Description rdf:about="http://geni-orca.renci.org/owl/42ad8a60-aa45-4d91-b27e-8b9a4fa12e1f#Link140">
    <rdf:type rdf:resource="http://geni-orca.renci.org/owl/topology.owl#NetworkConnection"/>
    <j.14:item rdf:resource="http://geni-orca.renci.org/owl/42ad8a60-aa45-4d91-b27e-8b9a4fa12e1f#StitchPort0"/>
    <j.14:item rdf:resource="http://geni-orca.renci.org/owl/ionradics.rdf#AL2S/UIUC/Cisco/6509/GigabitEthernet/1/1/1141-bb5b9994-e5f6-4a9b-8d09-dbc8d0c0bdc6/Stitching/Domain/intf/1f7e093b-e63f-4f01-bfb0-9fbaeef6aa68"/>
    <j.14:item rdf:resource="http://geni-orca.renci.org/owl/ionradics.rdf#ionradics/Domain/vlan/3f86dc21-4a04-462c-975e-e2161830b687/vlan"/>
    <j.14:item rdf:resource="http://geni-orca.renci.org/owl/disaNet.rdf#DisaNet/IBM/G8264/TenGigabitEthernet/1/0/ethernet-AL2S/Renci/Cisco/6509/TenGigabitEthernet/1/1/c1c9d5a9-cb55-4802-9474-44b52541baa7"/>
    <j.14:item rdf:resource="http://geni-orca.renci.org/owl/disaNet.rdf#disaNet/Domain/vlan/cf07376b-f7b6-4080-9f72-5c40532e5a1c/vlan"/>
    <j.14:item rdf:resource="http://geni-orca.renci.org/owl/disavmsite.rdf#Disa/ExoGeni/TenGigabitEthernet/1/ethernet-DisaNet/IBM/G8264/TenGigabitEthernet/1/1/ethernet/2aa635a3-f8b6-41df-b1dc-08ec9606a52f"/>
  </rdf:Description>


  <rdf:Description rdf:about="http://geni-orca.renci.org/owl/42ad8a60-aa45-4d91-b27e-8b9a4fa12e1f#StitchPort0">
    <rdf:type rdf:resource="http://geni-orca.renci.org/owl/topology.owl#Device"/>
    <j.4:inDomain rdf:resource="http://geni-orca.renci.org/owl/orca.rdf#bb5b9994-e5f6-4a9b-8d09-dbc8d0c0bdc6/Stitching/Domain"/>
    <j.15:hasURL>http://geni-orca.renci.org/owl/orca.rdf#bb5b9994-e5f6-4a9b-8d09-dbc8d0c0bdc6/Stitching/Domain</j.15:hasURL>
    <j.11:inRequestNetworkConnection rdf:resource="http://geni-orca.renci.org/owl/42ad8a60-aa45-4d91-b27e-8b9a4fa12e1f#Link140"/>
    <j.15:inConnection rdf:datatype="http://www.w3.org/2001/XMLSchema#boolean">true</j.15:inConnection>
  </rdf:Description>
```

**Comment by kthare10** [comment]
This issue results in following missing elements from manifest:

**Issue:1 : Link object has following properties missing:**
- interfaces
- GUID
- layer
- bandwidth

**Issue:2 : StitchPort object has following properties missing:**
- interfaces

**Root cause for the above two issues:**
For createSlice, requestModel is added to the manifest before any embedding logic is invoked or manifest is generated in RequestWorkflow::run(). So when ModifyHandler::createManifest() is invoked, element StitchPort and Link already exists and have interfaces, ModifyHandler::createManifest() just adds properties to the existing elements.

However, in case of modifySlice, requestModel is not added to the manifest. So when ModifyHandler::createManifest() is invoked, element StitchPort and Link do not exist, ModifyHandler::createManifest() only adds some properties to the existing elements and the properties which are assumed to be propagated from request are missing.

**Possible fix:**
ModifyHandler::createManifest() should add all the required properties for the elements instead of assuming them from being propagated from request.

**Issue:3 : Interface created has following properties missing:**
- GUID
- hostInterfaceName

**Root cause**: GUID and hostInterfaceName is added to an interface via ReservationConverter::setDependency() which is invoked only for createSlice or for added elements via modifySlice. For modified reservations, ReservationConverter::modifyReservations() is being invoked and this function does not set the GUI and hostInterfaceName.
**Possible Solution**: Instead of setting these properties in ReservationConverter, they should be set where Interface object is created i.e. CloudHandler::createInterface().

**Comment by kthare10** [comment]
**Problem Description:**
Adding first stitchport via modify slice is successful but results in broken manifest
All subsequent modifies fail

**Root Cause**
Manifest is missing properties for StitchPort, Link and interface object

Missing properties for StitchPort and Link Object
For createSlice, requestModel is added to the manifest before any embedding logic is invoked or manifest is generated in RequestWorkflow::run(). So when ModifyHandler::createManifest() is invoked, element StitchPort and Link already exists and have interfaces, ModifyHandler::createManifest() just adds properties to the existing elements.

However, in case of modifySlice, requestModel is not added to the manifest. So when ModifyHandler::createManifest() is invoked, element StitchPort and Link do not exist, ModifyHandler::createManifest() only adds some properties to the elements and the properties which are assumed to be propagated from request are missing.

**Missing properties for Interface object**
ReservationConverter::setDependency() which is invoked only for createSlice or for added elements via modifySlice. For modified reservations, ReservationConverter::modifyReservations() is being invoked and this function does not set the GUID and hostInterfaceName.

**Fix**
StitchPort and Link object in case of modify are being copied from the request
Interface object GUID is being set at the creation of interface instead of setDependency

**Additional Issues**
During the testing of the above fix, I noticed that deleting a StitchPort via modifySlice does not clear the controllerAssignedLabel set and manifest. This prevents re-adding a stitchport with same label.

**Fix**
Clearing slice local controllerAssignedLabel in SyncTags thread
Removed all the interfaces when updating manifest for a delete instead of removing only one interface

NOTE: This branch does not contain OSSRH changes. I am facing problems with running code with OSSRH changes with SOAP interface. I working to resolve that. I may have to submit this code without OSSRH changes if SOAP issue is not resolved.
